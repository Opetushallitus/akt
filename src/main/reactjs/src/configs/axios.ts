import axios from 'axios';

import { AppConstants } from 'enums/app';

const axiosInstance = axios.create();

axiosInstance.interceptors.request.use((config) => {
  config.headers = {
    ...config.headers,
    'Caller-Id': AppConstants.CalledID,
  };
  return config;
});

export default axiosInstance;
