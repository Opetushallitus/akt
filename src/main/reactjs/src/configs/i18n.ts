import i18n, { use, changeLanguage } from 'i18next';
import {
  initReactI18next,
  useTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import transFI from 'public/i18n/fi-FI.json';
import transSV from 'public/i18n/sv-SE.json';
import transEN from 'public/i18n/en-GB.json';
import langsFI from 'public/i18n/koodisto/langs/fi-FI.json';
import langsSV from 'public/i18n/koodisto/langs/sv-SE.json';
import langsEN from 'public/i18n/koodisto/langs/en-GB.json';

// Defaults and resources
const langFI = 'fi-FI';
const langSV = 'sv-SE';
const langEN = 'en-GB';

const detectionOptions = {
  order: ['localStorage', 'htmlTag'],
  caches: ['localStorage'],
};

const supportedLangs = [langFI, langSV, langEN];

const resources = {
  [langFI]: {
    translation: transFI,
    langs: langsFI,
  },
  [langSV]: {
    translation: transSV,
    langs: langsSV,
  },
  [langEN]: {
    translation: transEN,
    langs: langsEN,
  },
};

// Check types. It seems to be unused, but ts does some type checking with it.
declare module 'react-i18next' {
  interface CustomTypeOptions {
    defaultNS: typeof langFI;
    resources: {
      [langFI]: typeof transFI;
      [langSV]: typeof transSV;
      [langEN]: typeof transEN;
    };
  }
}

export const initI18n = () => {
  return use(initReactI18next)
    .use(LanguageDetector)
    .init({
      fallbackNS: 'langs',
      resources,
      detection: detectionOptions,
      fallbackLng: langFI,
      load: 'currentOnly',
      debug: process.env.NODE_ENV === 'development',
    });
};

export const useAppTranslation = (options: UseTranslationOptions<string>) => {
  // @ts-expect-error ts import fail
  return useTranslation(undefined, options);
};

export const getCurrentLang = (): string => {
  return i18n.language;
};

export const getSupportedLangs = (): string[] => {
  return supportedLangs;
};

export const changeLang = (language: string) => {
  return changeLanguage(language);
};

export const onLangChanged = (callback: (language: string) => void) => {
  return i18n.on('languageChanged', callback);
};
