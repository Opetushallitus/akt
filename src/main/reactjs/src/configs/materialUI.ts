import { createTheme } from '@mui/material/styles';

// Create Material UI theme configs
const primaryColor = '#FFFFFF';
const primaryLightColor = '#F5F5F5';
const primaryDarkColor = '#CCCCCC';
const secondaryColor = '#0A789C';
const secondaryLightColor = '#159ECB';
const secondaryDarkColor = '#00526C';
const primaryHeadingColor = '#2A2A2A';
const secondaryHeadingColor = '#FFFFFF';

const fontWeightBold = 700;
const fontWeightMedium = 500;
const fontWeightRegular = 400;

export const theme = createTheme({
  palette: {
    primary: {
      main: primaryColor,
      light: primaryLightColor,
      dark: primaryDarkColor,
    },
    secondary: {
      main: secondaryColor,
      light: secondaryLightColor,
      dark: secondaryDarkColor,
    },
    text: {
      primary: primaryHeadingColor,
      secondary: secondaryHeadingColor,
    },
  },
  typography: {
    htmlFontSize: 10,
    h1: {
      fontSize: '2.8rem',
      fontWeight: fontWeightMedium,
      lineHeight: '3.3rem',
      marginBottom: '2rem',
      color: primaryHeadingColor,
    },
    h2: {
      fontSize: '2rem',
      fontWeight: fontWeightMedium,
      lineHeight: '2.4rem',
      color: primaryHeadingColor,
    },
    h3: {
      fontSize: '1.6rem',
      fontWeight: fontWeightBold,
      lineHeight: '1.9rem',
      color: primaryHeadingColor,
    },
    body1: {
      fontSize: '1.6rem',
      fontWeight: fontWeightRegular,
      lineHeight: '2.4rem',
    },
  },
});
