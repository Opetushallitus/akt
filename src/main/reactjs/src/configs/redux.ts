import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

import configureStore from 'store/configureStore';

export const store = configureStore();

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
