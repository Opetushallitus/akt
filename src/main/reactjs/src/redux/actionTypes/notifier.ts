export const NOTIFIER_DIALOG_ADD = 'NOTIFIER_DIALOG/ADD';
export const NOTIFIER_DIALOG_REMOVE = 'NOTIFIER_DIALOG/REMOVE';

export const NOTIFIER_TOAST_ADD = 'NOTIFIER_TOAST/ADD';
export const NOTIFIER_TOAST_REMOVE = 'NOTIFIER_TOAST/REMOVE';

export const NOTIFIER_ACTION_DO_NOTHING = 'NOTIFIER_ACTION/DO_NOTHING';

// Contact Request Actions
export const NOTIFIER_ACTION_CONTACT_REQUEST_RESET =
  'NOTIFIER_ACTION_CONTACT_REQUEST/RESET';
export const NOTIFIER_ACTION_CONTACT_REQUEST_EMPTY =
  'NOTIFIER_ACTION_CONTACT_REQUEST/EMPTY';
