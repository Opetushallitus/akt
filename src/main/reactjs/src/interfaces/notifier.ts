import { Action } from 'redux';

import { Severity, NotifierTypes, Variant } from 'enums/app';

export interface NotifierButtonAction {
  title: string;
  variant: `${Variant}`;
  action: string;
  payload?: unknown;
}

export interface Notifier {
  id: string;
  severity: `${Severity}`;
  title: string;
  description: string;
  timeOut?: number;
  actions?: Array<NotifierButtonAction>;
}

export interface Toast extends Notifier {
  type: NotifierTypes.Toast;
}

export interface Dialog extends Notifier {
  type: NotifierTypes.Dialog;
}

export interface NotifierState {
  dialogs: Array<Dialog>;
  toasts: Array<Toast>;
}

export interface NotifierAction extends Action {
  id?: string;
  notifier?: Notifier;
  toasts?: Notifier;
}
