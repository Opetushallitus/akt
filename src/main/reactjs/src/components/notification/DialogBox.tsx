import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@mui/material';

import { Text } from 'components/elements/Text';
import { useAppDispatch, useAppSelector } from 'configs/redux';
import { notificationSelector } from 'redux/selectors/notifier';
import {
  executeNotifierAction,
  removeNotifierDialog,
} from 'redux/actions/notifier';

export const DialogBox = () => {
  // Redux
  const dispatch = useAppDispatch();
  const { dialogs } = useAppSelector(notificationSelector);
  const [dialog] = dialogs;

  const showDialog = dialogs?.length > 0;

  const handleDialogClose = () => {
    dispatch(removeNotifierDialog(dialog?.id));
  };

  const dispatchAction = (action: string) => {
    dispatch(executeNotifierAction(action));
    handleDialogClose();
  };

  return (
    <>
      {showDialog && (
        <Dialog
          className={`dialog-box--${dialog?.severity}`}
          open={showDialog}
          onClose={handleDialogClose}
        >
          <DialogTitle>{dialog?.title}</DialogTitle>
          <DialogContent>
            <Text>{dialog?.description}</Text>
          </DialogContent>
          <DialogActions>
            {dialog?.actions?.map((a, i) => (
              <Button
                key={i}
                variant={a.variant}
                color="secondary"
                onClick={() => dispatchAction(a.action)}
              >
                {a.title}
              </Button>
            ))}
          </DialogActions>
        </Dialog>
      )}
    </>
  );
};
