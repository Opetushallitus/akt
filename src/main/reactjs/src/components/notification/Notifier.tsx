import { Toast } from 'components/notification/Toast';
import { DialogBox } from 'components/notification/DialogBox';

export const Notifier = () => (
  <>
    <DialogBox />
    <Toast />
  </>
);
