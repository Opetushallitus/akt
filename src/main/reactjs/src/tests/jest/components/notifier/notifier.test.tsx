import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { DialogBox } from 'components/notification/DialogBox';
import { Toast } from 'components/notification/Toast';
import {
  selectorState,
  toastsArray,
  dialogsArray,
} from 'tests/jest/__fixtures__/notifier';
import { NotifierState } from 'interfaces/notifier';
import { useAppSelector } from 'configs/redux';
import { notifierReducer } from 'redux/reducers/notifier';
import {
  removeNotifierDialog,
  removeNotifierToast,
  showNotifierDialog,
  showNotifierToast,
} from 'redux/actions/notifier';

beforeEach(() => {
  const appSelector = useAppSelector as jest.Mock<NotifierState>;
  appSelector.mockImplementation(() => selectorState);
});

describe('DialogBox', () => {
  it('should render DialogBox correctly', () => {
    const { baseElement } = render(<DialogBox />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should show a Dialog when an action is dispatched', () => {
    const [dialog] = dialogsArray;
    const previousState = { dialogs: [], toasts: [] };
    const dialogCreator = showNotifierDialog(dialog);

    const newState = notifierReducer(previousState, dialogCreator);

    expect(newState).toEqual({ dialogs: [dialog], toasts: [] });
  });

  it('should remove a Dialog when an action is dispatched', () => {
    const [dialog] = dialogsArray;
    const previousState = { dialogs: [dialog], toasts: [] };
    const dialogCreator = removeNotifierDialog(dialog.id);

    const newState = notifierReducer(previousState, dialogCreator);

    expect(newState).toEqual({ dialogs: [], toasts: [] });
  });
});

describe('Toast', () => {
  it('should render Toast correctly', () => {
    const tree = renderer.create(<Toast />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show a Toast when an action is dispatched', () => {
    const [toast] = toastsArray;
    const previousState = { dialogs: [], toasts: [] };
    const toastCreator = showNotifierToast(toast);

    const newState = notifierReducer(previousState, toastCreator);

    expect(newState).toEqual({ dialogs: [], toasts: [toast] });
  });

  it('should remove a Toast when an action is dispatched', () => {
    const [toast] = toastsArray;
    const previousState = { dialogs: [], toasts: [toast] };
    const toastCreator = removeNotifierToast(toast.id);

    const newState = notifierReducer(previousState, toastCreator);

    expect(newState).toEqual({ dialogs: [], toasts: [] });
  });
});
