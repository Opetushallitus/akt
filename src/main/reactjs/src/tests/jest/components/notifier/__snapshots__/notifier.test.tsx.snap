// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogBox should render DialogBox correctly 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root dialog-box--info css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="mui-1"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id="mui-1"
        >
          Test Title 1
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Test Description 1
          </p>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Test Action 1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Toast should render Toast correctly 1`] = `null`;
